### PR TITLE
users can have spaces in urls now

### LINKED ISSUE #121: Bug: handle white space in URL
It looks like the URL is still valid when it contains white space, which leads to issue when browser sees it. Shouldn't be a hard fix but there seem to be multiple approach to it.

image
image

### PR DESCRIPTION
**Pull request by WhoaNellie**
Encoding the URI in a few places so users will be able to put spaces in their links

### CODE REVIEW

File: client/src/pages/Scheduler.js [hunk 1]
```diff
@@ -54,9 +54,11 @@ const Scheduler = () => {
 
   useEffect(() => {
     // Matching only the first two params so /hostname/eventname/datelinks will still work correctly
-    const regex = /appt\/([0-9a-z-]*\/[0-9a-z-]*)/g;
+    const regex = /appt\/([%\s0-9a-z-]*\/[%\s0-9a-z-]*)/g;
```
Comment (alvyjudy) [inline client/src/pages/Scheduler.js:57]: I wonder if we should just grab the string and run a `decodeURIComponent()` on it. But if this works then let's just go with it.

File: client/src/components/SelectTabs/EventTypesTab.js [hunk 2]
```diff
@@ -296,7 +296,7 @@ export default function EventTypesTab() {
       .post("/api/event", {
         ...eventBody,
         duration: minutes,
-        link: `${userURL}/${eventBody.link}`,
+        link: encodeURI(`${userURL}/${eventBody.link}`),
```
Comment (alvyjudy) [inline client/src/components/SelectTabs/EventTypesTab.js:299]: The proper function should be encodeURIComponent I think. But this should work as well.

**Review by alvyjudy — APPROVED** [review]
I left some suggestions but they aren't critical. If it works let's go with what you have already!

File: client/src/pages/Scheduler.js (hunk 1 above)
Comment (WhoaNellie) [reply client/src/pages/Scheduler.js:57]: This works but that could possibly be a better way!

File: client/src/components/SelectTabs/EventTypesTab.js (hunk 2 above)
Comment (WhoaNellie) [reply client/src/components/SelectTabs/EventTypesTab.js:299]: Link isn't a complete uri, it's just the user's url pref and suffix so this should be fine I believe!

**Pull request merged**